pkg/services/nvd.go

```diff
@@ -51,45 +51,29 @@ const (
 )
 
 func fetchNvdDataByCPE(cpe string, baseNvdAPIURL string) (*dto.NvdAPIResponse, error) {
-	// Use custom http client with a timeout
-	client := createNVDHTTPClient()
-
-	// Build URL
-	query := url.Values{}
-	query.Set("cpeName", cpe)
-	query.Set("isVulnerable", "")
-	apiURL := baseNvdAPIURL + "?" + query.Encode()
-
-	var nvdResponse *dto.NvdAPIResponse
-	var err error
-
-	for attempt := 0; attempt <= maxRetries; attempt++ {
-		nvdResponse, err = attemptFetch(client, apiURL)
-
-		// Success case
-		if err == nil {
-			return nvdResponse, nil
+	var currentResponse *dto.NvdAPIResponse
```

Comment:
To improve code readability, this variable should be re-named to `aggregatedResponse`, since it's the aggregation of all NVD responses for a particular CPE.
```suggestion
	var aggregatedResponse *dto.NvdAPIResponse
```